gemrb/core/GUI/GameControl.cpp

```diff
@@ -2038,6 +2028,33 @@ bool GameControl::HandleActiveRegion(InfoPoint *trap, Actor * actor, const Point
 	return false;
 }
 
+// Calculate the angle between a clicked point and the first selected character,
+// so that we can set a sensible orientation for the formation
+void GameControl::InitFormation (const Point &clickPoint)
+{
+	/* We get called on initial click, but also on drag.  Perform the setup only once.  */
+	if (isFormationRotation)
+		return;
+
+	if (core->GetGame()->selected.size() == 0) {
+		isFormationRotation = false;
+		return;
+	}
+
+	isFormationRotation = true;
+	Game* game = core->GetGame();
+	int max = game->GetPartySize(false);
+	formationBaseAngle = 0;
+	for (int idx = 1; idx <= max; idx++) {
+		Actor *act = game->FindPC(idx);
+		if (act->IsSelected()) {
```

Comment:
What happens if no pc is selected (basically `OnlyNPCsSelected()`)? Can happen if you start the rotation with only summons. In that sense it would be better to iterate over `selected` from Game.

Reply:
Hmm, in that case we get the old behaviour. I'll change it to use the code from PerformSelectedAction instead.

Reply:
Not sure what you mean, but let's see.